### ISSUE TITLE
Router Provider does not support navigate from ReactRouter v7

### ISSUE THREAD

**Issue by IWI013** [adobe/react-spectrum#7578]
### Provide a general summary of the issue here

In react router v7, NavigateFunction is a function returning void or Promise<void>. However the current router provider from react-aria-components only supports NavigateFunction to be returning void like in v6. This blocks me from upgrading to the newest version of react router.

https://github.com/remix-run/react-router/commit/361eb79548bae00de211222c80234e7250eb26d9#diff-0084431f6040dd28c2c3948d556762903dcdeb348b9bb28c8fbe01524c9070a3R158-R161

### 🤔 Expected Behavior?

I should be able to use navigate from react router v7 with the router provider from react-aria-components 

### 😯 Current Behavior

I can not use navigate from react router v7 with the router provider from react-aria-components  without errors

### 💁 Possible Solution

Support the newest version of NavigateFunction from react router v7 which returns void or Promise<void>

### 🖥️ Steps to Reproduce

1. Pass navigate using useNavigate from react router v7 to the RouterProvider

### Version

1.5.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Mac 15

**Comment by devongovett** [comment]
Can you just wrap it in another function to avoid the typescript issue?

```jsx
let navigate = useNavigate();

<RouterProvider navigate={to => navigate(to)}>
  {/* ... */}
</RouterProvider>
```

**Comment by IWI013** [comment]
Didn't think of that, thanks. Although in my opinion they should be the same types
